src/main/java/ruproject/domain/Contenido.java

```diff
@@ -22,5 +22,6 @@ public class Contenido {
     @Column
```

Comment:
1. Tienes que revisar como estos datos son representados en la BD, posiblemente como es una BD relacional, puede que algunas de estas entidades, pueda ser mejor representada en una tabla. Por ejemplo, si un libro puede ser material para dos materias distintas.
2. La anotación en estos casos no es necesaria, si no le especificas a la columna un tamaño, o un nombre diferente al que se le asignará por defecto. Para mayor información: https://stackoverflow.com/questions/22735543/column-annotation-absolutly-necessary-in-jpa